ampd/src/handlers/starknet_verify_msg.rs

```diff
@@ -47,65 +47,53 @@ struct PollStartedEvent {
     participants: Vec<TMAddress>,
 }
 
-pub struct Handler<C, B>
+pub struct Handler<C>
 where
     C: StarknetClient,
-    B: BroadcasterClient,
 {
     worker: TMAddress,
     voting_verifier: TMAddress,
     rpc_client: C,
-    broadcast_client: B,
     latest_block_height: Receiver<u64>,
 }
 
-impl<C, B> Handler<C, B>
+impl<C> Handler<C>
 where
     C: StarknetClient + Send + Sync,
-    B: BroadcasterClient,
 {
     pub fn new(
         worker: TMAddress,
         voting_verifier: TMAddress,
         rpc_client: C,
-        broadcast_client: B,
         latest_block_height: Receiver<u64>,
     ) -> Self {
         Self {
             worker,
             voting_verifier,
             rpc_client,
-            broadcast_client,
             latest_block_height,
         }
     }
 
-    async fn broadcast_votes(&self, poll_id: PollId, votes: Vec<Vote>) -> Result<()> {
-        let msg = serde_json::to_vec(&ExecuteMsg::Vote { poll_id, votes })
-            .expect("vote msg should serialize");
-        let tx = MsgExecuteContract {
+    fn vote_msg(&self, poll_id: PollId, votes: Vec<Vote>) -> MsgExecuteContract {
+        MsgExecuteContract {
             sender: self.worker.as_ref().clone(),
             contract: self.voting_verifier.as_ref().clone(),
-            msg,
+            msg: serde_json::to_vec(&ExecuteMsg::Vote { poll_id, votes })
+                .expect("vote msg should serialize"),
             funds: vec![],
-        };
-
-        self.broadcast_client
-            .broadcast(tx)
-            .await
-            .change_context(Error::Broadcaster)
+        }
     }
 }
 
 #[async_trait]
-impl<V, B> EventHandler for Handler<V, B>
+impl<V> EventHandler for Handler<V>
 where
     V: StarknetClient + Send + Sync,
-    B: BroadcasterClient + Send + Sync,
 {
     type Err = Error;
 
-    async fn handle(&self, event: &events::Event) -> Result<()> {
+    async fn handle(&self, event: &events::Event) -> Result<Vec<Any>> {
```

Comment:
A `Vec<Any>` ? Wow, i'm curious about the reasoning behind this change. Sounds like a placeholder for a future change ? Well, looks like we would need to do the same in the Solana part :pencil2: . Thanks for making it so explicit, it was easy to find :mag: 

cc: @roberts-pumpurs @tilacog

Reply:
I didn't put much thought into it to be honest. I was aiming at making it compile :D 
You can add it as a question to them for this week's meeting if you like?

Reply:
Just a note - Another major change is that broadcast client has been removed and all `worker` occurrences seem to have been changed to `verifier`. Even the `bond-worker` command is changed to `bond-verifier`.

Reply:
Ok i think now i understand better this changes (i am adopting them in sol part). They just removed the broadcaster client _probably_ because is a constant dependency in all the handlers, so it was moved one abstraction level up, leaving the broadcasting responsibility to the core ampd core. That way implementers only need to care about its own deps and return the votes in the handling interface. 

I generally consider this a good move.